beacon_node/lighthouse_network/src/rpc/active_requests_limiter.rs

```diff
@@ -0,0 +1,121 @@
+use crate::rpc::{Protocol, SubstreamId};
+use libp2p::swarm::ConnectionId;
+use libp2p::PeerId;
+use std::collections::hash_map::Entry;
+use std::collections::HashMap;
+
+/// Restricts more than two inbound requests from running simultaneously on the same protocol per peer.
+pub(super) struct ActiveRequestsLimiter {
+    requests: HashMap<PeerId, Vec<(Protocol, ConnectionId, SubstreamId)>>,
+}
+
+impl ActiveRequestsLimiter {
+    pub(super) fn new() -> Self {
+        Self {
+            requests: HashMap::new(),
+        }
+    }
+
+    /// Allows if there is not a request on the same protocol.
+    pub(super) fn allows(
+        &mut self,
+        peer_id: PeerId,
+        protocol: Protocol,
+        connection_id: &ConnectionId,
+        substream_id: &SubstreamId,
+    ) -> bool {
+        match self.requests.entry(peer_id) {
+            Entry::Occupied(mut entry) => {
+                for (p, _cid, _sid) in entry.get_mut().iter_mut() {
+                    // Check if there is a request on the same protocol.
+                    if p == &protocol {
+                        return false;
+                    }
+                }
+
+                // Request on the same protocol was not found.
+                entry
+                    .get_mut()
+                    .push((protocol, *connection_id, *substream_id));
+                true
+            }
+            Entry::Vacant(entry) => {
+                // No active requests for the peer.
+                entry.insert(vec![(protocol, *connection_id, *substream_id)]);
+                true
+            }
+        }
+    }
```

Comment:
since this function is now just this, wdyt of having this function at the `Behaviour` level, removing `ActiveRequestsLimiteR` and therefore eliminating  the need for a duplicated `HashMap` with requests?

Reply:
That makes sense. 💡 Thanks!

Reply:
Removed ActiveRequestsLimiter in https://github.com/sigp/lighthouse/pull/5923/commits/3c058b3bacdc378721d1c6faf2a4d65c5b77725e